### PR TITLE
8322806: Eliminate -Wparentheses warnings in aarch64 code

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to eliminate some -Wparentheses warnings.  This
involved simply adding a few parentheses to make some implicit operator
precedence explicit.

Testing: mach5 tier1

Also ran mach5 tier1 with these changes in conjunction enabling -Wparentheses
and other changes needed to make that work.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322806](https://bugs.openjdk.org/browse/JDK-8322806): Eliminate -Wparentheses warnings in aarch64 code (**Enhancement** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [82e75c86](https://git.openjdk.org/jdk/pull/17210/files/82e75c86f8f4da96fc55a45c51b6d0209858c3cb)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [82e75c86](https://git.openjdk.org/jdk/pull/17210/files/82e75c86f8f4da96fc55a45c51b6d0209858c3cb)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17210/head:pull/17210` \
`$ git checkout pull/17210`

Update a local copy of the PR: \
`$ git checkout pull/17210` \
`$ git pull https://git.openjdk.org/jdk.git pull/17210/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17210`

View PR using the GUI difftool: \
`$ git pr show -t 17210`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17210.diff">https://git.openjdk.org/jdk/pull/17210.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17210#issuecomment-1873582858)

### CODE REVIEW

**Review by stefank — APPROVED** [review]
Seems fine. FWIW, the second usage of `ret` looks redundant.

**Review by dholmes-ora — APPROVED** [review]
LGTM. Thanks

File: src/hotspot/cpu/aarch64/frame_aarch64.inline.hpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ inline bool frame::equal(frame other) const {
               && unextended_sp() == other.unextended_sp()
               && fp() == other.fp()
               && pc() == other.pc();
-  assert(!ret || ret && cb() == other.cb() && _deopt_state == other._deopt_state, "inconsistent construction");
+  assert(!ret || (ret && cb() == other.cb() && _deopt_state == other._deopt_state), "inconsistent construction");
```
[the remark this answers is not in the document]
Comment (stefank) [reply src/hotspot/cpu/aarch64/frame_aarch64.inline.hpp:198]: I agree:
https://github.com/openjdk/jdk/pull/17210#pullrequestreview-1800130145

[the remark this answers is not in the document]
Comment (kimbarrett) [reply src/hotspot/cpu/aarch64/frame_aarch64.inline.hpp]: Agreed, and updated.

**Comment by kimbarrett** [comment]
Thanks for reviews @dholmes-ora , @stefank , and comment from @shipilev .

**Pull request closed without merging**